### PR TITLE
Transcode videos that do not support direct play

### PR DESCRIPTION
**Pull request by thomabx**
**Changes**
Checks device to see if it supports direct play. If direct play is not supported, creates a session to transcode to h264 and acc.

### CODE REVIEW

**Review by bisby — CHANGES_REQUESTED** [review]
This is awesome, and I'm going to click approve.

But first, could we clean up the whitespace on this?

There's some trailing whitespace, mixed indentation, and an indented blank line.

After that. :+1: :100:

**Comment by thomabx** [comment]
I think I got all the formatting and whitespace corrected now.

File: source/VideoPlayer.brs [hunk 1]
```diff
@@ -17,35 +17,44 @@ function VideoContent(id) as object
...
   if server_is_https() then
     content.setCertificatesFile("common:/certs/ca-bundle.crt")
   end if
-
   return content
 end function
 
+function directPlaySupported(meta as object) as boolean
+    devinfo = CreateObject("roDeviceInfo")
+	return devinfo.CanDecodeVideo({ Codec: meta.json.MediaStreams[0].codec }).result
```
Comment (bisby) [inline source/VideoPlayer.brs]:
```suggestion
    return devinfo.CanDecodeVideo({ Codec: meta.json.MediaStreams[0].codec }).result
```

File: source/api/Items.brs [hunk 2]
```diff
@@ -22,7 +22,18 @@ function UserItemsResume(params = {} as object)
...
 end function
 
-
+function ItemGetSession(id as string)
+  params = {
+    UserId: get_setting("active_user"),
+    StartTimeTicks: "0",
+    IsPlayback: "true",
+    AutoOpenLiveStream: "true",
+    MaxStreamingBitrate: "140000000"
```
Comment (bisby) [inline source/api/Items.brs]:
```suggestion
    MaxStreamingBitrate: "140000000"
```

**Review by bisby — APPROVED** [review]
Thanks for your patience!

**Pull request merged**